test/move.js

```diff
@@ -0,0 +1,77 @@
+'use strict';
+
+var assert = require('assert');
+var path = require('path');
+var sinon = require('sinon');
+var editor = require('..');
+var memFs = require('mem-fs');
+
+describe('#move()', function () {
+  beforeEach(function() {
+    var store = memFs.create();
+    this.fs = editor.create(store);
+  });
+
+  it('move file', function () {
+    var filepath = path.join(__dirname, 'fixtures/file-a.txt');
+    var initialContents = this.fs.read(filepath);
+    var newpath = '/new/path/file.txt';
+    this.fs.move(filepath, newpath);
+    assert.equal(this.fs.read(newpath), initialContents);
+    assert.throws(this.fs.read.bind(this.fs, filepath));
+  });
+
+  it('move directory', function() {
+    var filename = 'file.txt';
+    var dirpath = path.join(__dirname, 'fixtures/nested');
+    var filepath = path.join(dirpath, filename);
+    var newdirpath = '/new/path';
+    var newfilepath = path.join(newdirpath, filename);
+    this.fs.move(dirpath, newdirpath);
+    assert.equal(this.fs.store.get(newfilepath).state, 'modified');
+    assert.throws(this.fs.read.bind(this.fs, filepath));
+  });
+
+  it('move file to an existing `to` path', function() {
```

Comment:
Yes, but the overwrite could be unintentional, especially when it is done without giving any warning.